core/runtime/src/main/java/io/quarkus/runtime/Application.java

```diff
@@ -182,21 +184,18 @@ public static Application currentApplication() {
     public final void run(String[] args) {
         try {
             if (ImageInfo.inImageRuntimeCode() && System.getenv(DISABLE_SIGNAL_HANDLERS) == null) {
-                final SignalHandler handler = new SignalHandler() {
-                    @Override
-                    public void handle(final Signal signal) {
-                        System.exit(signal.getNumber() + 0x80);
-                    }
-                };
-                Signal.handle(new Signal("INT"), handler);
-                Signal.handle(new Signal("TERM"), handler);
-                Signal.handle(new Signal("HUP"), handler);
-                Signal.handle(new Signal("QUIT"), new SignalHandler() {
-                    @Override
-                    public void handle(final Signal signal) {
-                        DiagnosticPrinter.printDiagnostics(System.out);
-                    }
-                });
+                final SignalHandler handler = signal -> System.exit(signal.getNumber() + 0x80);
+                final SignalHandler quitHandler = signal -> DiagnosticPrinter.printDiagnostics(System.out);
```

Comment:
@gastaldi I will backport this as is but could you go back to not using lambdas? We are trying to avoid them in runtime code and AFAICS, this is runtime code.

Thanks!

Reply:
Sure, I'll provide a new PR without them, thank you!